### PR TITLE
Migrate to using ArgoCD operator

### PR DESCRIPTION
**Pull request by HappyTetrahedron**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -17,12 +18,10 @@ parameters:
       default: info
 
     images:
-      argocd:
-        image: quay.io/argoproj/argocd
-        tag: v2.3.12
-      redis:
-        image: docker.io/library/redis
-        tag: '6.2.6'
```
[the remark this answers is not in the document]
Comment (HappyTetrahedron) [reply class/defaults.yml:25]: I've noticed some inconsistencies; some places use `image` and `tag` while others use `registry`, `repository` and `tag`. Which should I use?

File: tests/golden/defaults/argocd/argocd/10_operator/apiextensions.k8s.io_v1_customresourcedefinition_applications.argoproj.io.yaml
[the remark this answers is not in the document]
Comment (HappyTetrahedron) [reply tests/golden/defaults/argocd/argocd/10_operator/apiextensions.k8s.io_v1_customresourcedefinition_applications.argoproj.io.yaml:1]: No, I'm not 100% sure. Fwiw I haven't observed any issues while testing.

The operator would ship with argocd v2.4.0 by default; I assume that's also the version of the CRDs then.

I've diffed the CRDs; in the `application` resource, the `ksonnet` source property disappears and a number of new optional parameters are introduced. In `appproject`, only new optional parameters appear, so there I don't see an issue.

I don't believe we use ksonnet? ^^

File: class/defaults.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (DebakelOrakel) [reply class/defaults.yml:25]: https://syn.tools/syn/explanations/commodore-components/container-images.html
`registry, repository, tag`

File: tests/golden/defaults/argocd/argocd/10_operator/apiextensions.k8s.io_v1_customresourcedefinition_applications.argoproj.io.yaml
[the remark this answers is not in the document]
Comment (simu) [reply tests/golden/defaults/argocd/argocd/10_operator/apiextensions.k8s.io_v1_customresourcedefinition_applications.argoproj.io.yaml:1]: We don't use ksonnet, so this should be fine then.

**Review by simu — APPROVED** [review]
LGTM overall, see inline comments for making the upgrade less painful.

**Review by DebakelOrakel — APPROVED** [review]

**Pull request merged**